iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -562,3 +562,22 @@ TEST_F(PortManager_test, PortDestroy)
         }
     }
 }
+
+TEST_F(PortManager_test, IntrospectionMemoryManagerNotInitialzed)
+{
+    IceOryxRouDiMemoryManager* m_roudiMemoryManagerTest{nullptr};
+    auto config = iox::RouDiConfig_t().setDefaults();
+    m_roudiMemoryManagerTest = new IceOryxRouDiMemoryManager(config);
+
+    auto result = m_roudiMemoryManagerTest->introspectionMemoryManager();
+    EXPECT_THAT(result, Eq(iox::cxx::nullopt_t()));
+
+    auto resultTest = m_roudiMemoryManagerTest->segmentManager();
+    EXPECT_THAT(result, Eq(iox::cxx::nullopt_t()));
+
+    auto testResult = m_roudiMemoryManagerTest->portPool();
+    EXPECT_THAT(testResult, Eq(iox::cxx::nullopt_t()));
```

Comment:
There is in file port_pool_memory_block.cpp named as well portPool() but it seems to never be called. Compare with introspectionMemoryManager() and segmentManager(), those functions are calling method in files mempool_collection_memory_block.cpp and mempool_segment_manager_memory_block.cpp. Why portPool() from iceoryxRoudiMananger is not doing the same ?

Reply:
This might be a bug. Right now I'm wondering why this is working at all. There was quite a big refactoring two months ago, maybe something went wrong. I have to investigate this more closely tomorrow.

Reply:
okay, I got it. The `portPool` method from `port_pool_memory_block.cpp` should probably be called `portPoolData`. It's called in `IceOryxRouDiMemoryManager::createAndAnnounceMemory`, where the actual `PortPool` is created. The `IceOryxRouDiMemoryManager::portPool()` return the actual `PortPool` pointer.